src/main/kotlin/snyk/common/intentionactions/AlwaysAvailableReplacementIntentionAction.kt

```diff
@@ -8,19 +8,28 @@ import com.intellij.openapi.util.TextRange
 import com.intellij.psi.PsiFile
 import com.intellij.util.DocumentUtil
 import io.snyk.plugin.refreshAnnotationsForOpenFiles
+import io.snyk.plugin.services.SnykAnalyticsService
 import io.snyk.plugin.ui.SnykBalloonNotificationHelper
+import snyk.analytics.QuickFixIsDisplayed
+import snyk.analytics.QuickFixIsTriggered
 
 class AlwaysAvailableReplacementIntentionAction(
     val range: TextRange,
     val replacementText: String,
     private val intentionText: String = intentionDefaultTextPrefix,
     private val familyName: String = intentionDefaultFamilyName,
-    val message: String = ""
+    val message: String = "",
+    val analyticsService: SnykAnalyticsService?
```

Comment:
could have default value and reassigned just for tests

Reply:
I like that!